lambdas/issue-client-access-token/src/main/java/uk/gov/di/ipv/core/issueclientaccesstoken/validation/TokenRequestValidator.java

```diff
@@ -30,25 +31,29 @@ public class TokenRequestValidator {
     private static final Logger LOGGER = LogManager.getLogger();
 
     private final ConfigService configService;
-
     private final ClientAuthJwtIdService clientAuthJwtIdService;
-    private ClientAuthenticationVerifier<Object> verifier;
+    private final OAuthKeyService oAuthKeyService;
+
+    private ClientAuthenticationVerifier<Object> clientAuthVerifier;
 
     public TokenRequestValidator(
-            ConfigService configService, ClientAuthJwtIdService clientAuthJwtIdService) {
+            ConfigService configService,
+            ClientAuthJwtIdService clientAuthJwtIdService,
+            OAuthKeyService oAuthKeyService) {
         this.configService = configService;
         this.clientAuthJwtIdService = clientAuthJwtIdService;
+        this.oAuthKeyService = oAuthKeyService;
     }
 
     public void authenticateClient(String requestBody) throws ClientAuthenticationException {
-        if (verifier == null) {
-            this.verifier = getClientAuthVerifier(configService);
+        if (clientAuthVerifier == null) {
+            this.clientAuthVerifier = getClientAuthVerifier(configService);
```

Comment:
This is a bit weird, but I think it's like this so that the audience config value doesn't get baked in to the snapstart image.